include/arch/x86/linker.ld

```diff
@@ -69,6 +69,20 @@ SECTIONS
 
 #include <linker/rel-sections.ld>
 
+    /*
+     * The .plt and .iplt are here according to
+     * 'i586-zephyr-elf-ld --verbose', before text section.
+     */
+    SECTION_PROLOGUE(.plt,,)
```

Comment:
My understanding is that .plt is used only for dynamic linking, which we don't support.

https://reverseengineering.stackexchange.com/questions/1992/what-is-plt-got

Maybe just drop these sections instead?